### PR TITLE
Migrate field templates to field vue components

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR is based on https://github.com/matomo-org/tag-manager/pull/407.**

Changes:
* Migrate field-textarea-variable-template to FieldTextareaVariable.
* Migrate field-variable-template.html to FieldTemplateVariable.
* Migrate field-variabletype-template.html to FieldVariableTypeTemplate.
* Change references to the old templates to the new vue components

Note: the old templates and templateFile checks still exist for backwards compatibility.

image

image

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: Template/BaseTemplate.php [hunk 1]
```diff
@@ -27,9 +27,25 @@ abstract class BaseTemplate
 
     protected $templateType = '';
 
+    /**
+     * @deprecated
```
Comment (AltamashShaikh) [inline Template/BaseTemplate.php]: can we add what to be used instead of this constant 
Eg: @deprecated Use FIELD_VARIABLE_COMPONENT instead

**Review by AltamashShaikh — CHANGES_REQUESTED** [review]
@diosmosis  I see the configure variable icon position is changed as well as alignment looks broken to me
**Existing**
Screenshot from 2022-02-22 08-37-28

**Changed**
Screenshot from 2022-02-22 08-37-20

**Comment by diosmosis** [comment]
applied the fixes, not sure if the screenshots need to be updated

**Review by AltamashShaikh — APPROVED** [review]

**Review by AltamashShaikh — APPROVED** [review]
@diosmosis Need to update the test cases, rest looks fine
Feel free to merge once test cases are fixed

**Pull request merged**